sklearn/model_selection/_search.py

```diff
@@ -418,7 +452,8 @@ def score(self, X, y=None):
         -------
         score : float
         """
-        self._check_is_fitted("score")
+        _check_refit(self, "score")
```

Comment:
On `main`:

- If `refit=False`, then `NotFittedError` is raised stating that `refit=False`
- If `refit=True`, then `check_is_fitted` is called.

With this PR, I think the behavior is still the same.

Reply:
I just replaced the `NotFittedError` with an `AttributeError` since this is indeed the issue.
In terms of behaviour, this is the exact current behaviour.

Reply:
Have you pushed the changes @glemaitre ?

Reply:
I accepted the suggestions of @thomasjpfan but as I mentioned, I don't think that there is a change of behavior here. Which backward incompatibility do you observe (I might have overlooked as well since I was using the tests to ensure that).